### PR TITLE
Better equals check

### PR DESCRIPTION
**Pull request by avently**
#### What is it?
- [x] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
It ensures that queues are not the same. Without this check when you have multiple videos in the backstack and navigating back via Back button you'll get duplicated videos

#### Fixes the following issue(s)

Don't know. To hard to search for "duplicate" keyword:)

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java [hunk 1]
```diff
@@ -528,7 +528,16 @@ public boolean equals(@Nullable final Object obj) {
             return false;
         }
         final PlayQueue other = (PlayQueue) obj;
-        return streams.equals(other.streams);
+        if (size() != other.size()) {
+            return false;
+        }
+        for (int i = 0; i < size(); i++) {
+            // Check is based on URL
+            if (!streams.get(i).getUrl().equals(other.streams.get(i).getUrl())) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java:536]: I know that the url should technically be enough to uniquely identify a stream, but in other places when comparing streams the service id is also used, so I'd add it.

**Review by Stypox — CHANGES_REQUESTED** [review]
Other than the small comment I made, this looks good to me

File: app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java (hunk 1 above)
Comment (avently) [reply app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java]: Added this check

**Comment by avently** [comment]
You can merge this PR if it's ok, then I'll merge dev into #6895 and it will make testing of #6895 much easier

**Review by Stypox — APPROVED** [review]
Thank you! I'm going to merge this in the afternoon, after having created a separate branch on dev for an almost-hotfix release 0.22.1

**Pull request merged**